packages/ti_rapid7_threat_command/data_stream/ioc/elasticsearch/ingest_pipeline/default.yml

```
@@ -264,6 +264,38 @@ processors:
       ignore_empty_value: true
       if: ctx.threat?.indicator?.geo?.country_iso_code == null
       ignore_failure: true
+  - set:
+      field: threat.indicator.name
+      copy_from: threat.indicator.url.original
+      ignore_empty_value: true
+      if: ctx.threat?.indicator?.url?.original != null && ctx.threat.indicator.type == 'url'
+  - set:
+      field: threat.indicator.name
+      copy_from: threat.indicator.url.domain
+      ignore_empty_value: true
+      if: ctx.threat?.indicator?.url?.domain != null && ctx.threat.indicator.type == 'domain-name'
+  - set:
+      field: threat.indicator.name
+      copy_from: threat.indicator.ip
+      ignore_empty_value: true
+      if: ctx.threat?.indicator?.ip != null && ['ipv4-addr', 'ipv6-addr'].contains(ctx.threat.indicator.type)
```

Prefer non-allocating approach.
```suggestion
      if: >-
        ctx.threat?.indicator?.ip != null && (
          ctx.threat.indicator.type == 'ipv4-addr' ||
          ctx.threat.indicator.type == 'ipv6-addr'
        )
```

---

Makes sense 👍🏼 updated as suggested